### ISSUE TITLE
Uploaded channel avatar photo doesn't appear

### ISSUE THREAD

**Issue by csimpi** [RocketChat/Rocket.Chat#22903]
### Description:

When I upload an avatar for a channel, the uploaded photo appears on the upload section but nowhere else.

### Steps to reproduce:

1. Go to Room Information page
2. Click on Edit button
3. Upload a new avatar
4. Save

### Expected behavior:

The new avatar should appear, the default avatar should be replaced

### Actual behavior:

The default avatar stays everywhere.

### Server Setup Information:

- Version of Rocket.Chat Server: 3.17.0
- Operating System: Linux 
- Deployment Method:  tar
- Number of Running Instances: 1
- DB Replicaset Oplog: 
- NodeJS Version:  12.22.1
- MongoDB Version: 4.4.3 / wiredTiger  (oplog Enabled)

### Client Setup Information

- Desktop App or Browser Version: chrome newest
- Operating System: Linux Debian 20.4

**Comment by ankar84** [comment]
Related to https://github.com/RocketChat/Rocket.Chat/issues/22307
Could you check if avatar set in mobile client when you set it in web?

**Comment by csimpi** [comment]
> Related to #22307
> Could you check if avatar set in mobile client when you set it in web?

Yes, it sets in right away on mobile, when I set it on web. On web, nothing happens.

**Comment by ankar84** [comment]
> Yes, it sets in right away on mobile, when I set it on web. On web, nothing happens.

I have exactly the same experience.
So, thinking logically - avatar is set properly  if we see it in mobile client. 
But the real problem is in web client - it's now updating avatar right after it was set. Or not show properly.

**Comment by debdutdeb** [comment]
Thanks Peter ! Thanks Anton 👋🏼 

I'll try and get this fixed ASAP. Should be an easy one.

**Comment by ankar84** [comment]
> Thanks Peter ! Thanks Anton 👋🏼
> 
> I'll try and get this fixed (pass to the product team) ASAP. Should be an easy one.

Still not fixed in 3.18.1

**Comment by pierreozoux** [comment]
I have a slightly similar, but maybe a bit different.

It doesn't reproduce consistently, but If I try 2 times, usually, I can reproduce.
We use an s3 compatible api as an object store (minio) and the logs look good at the upload:
Server:
```
{"level":35,"time":"2022-03-30T13:11:04.017Z","pid":1,"hostname":"chat-xx-app-xx","name":"API","method":"POST","url":"/api/v1/rooms.saveRoomSettings","userId":"xx","userAgent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:96.0) Gecko/xx Firefox/96.0","length":"15389","host":"chat.liiib.re/","referer":"https://chat.liiib.re/group/testpierre/channel-settings","remoteIP":"xx","status":200,"responseTime":7}
```
And in the browser:
![Screenshot from 2022-03-30 15-15-36](https://user-images.githubusercontent.com/1863453/160843291-567eeb77-4f31-4e1c-9961-2bd049ecfcfe.png)


But still, then I get a 404 on the image that should have been uploaded:
![Screenshot from 2022-03-30 15-16-33](https://user-images.githubusercontent.com/1863453/160843448-8d6c1f00-dd47-4ae5-aecd-d23df2635d7d.png)
And if I actually check on my object store, I don't see the mentioned file.

I checked also the logs on the object store backend, and the behavior is strange:
```
xx - - [30/Mar/2022:13:21:27 +0000] "PUT /chat-liiib-re/xx/avatars/room-wZQ7J42Y5SdMquaJ5 HTTP/1.1" 200 25 "-" "aws-sdk-nodejs/2.738.0 linux/v14.18.3 callback" 11200 0.026 [minio-hot-minio-443] [] 10.244.30.17:9000 0 0.024 200 xx
xx - - [30/Mar/2022:13:21:27 +0000] "GET /chat-liiib-re/xx/avatars/room-wZQ7J42Y5SdMquaJ5 HTTP/1.1" 200 10626 "-" "aws-sdk-nodejs/2.738.0 linux/v14.18.3" 491 0.015 [minio-hot-minio-443] [] 10.244.68.231:9000 10626 0.012 200 xx
xx - - [30/Mar/2022:13:21:28 +0000] "DELETE /chat-liiib-re/xx/avatars/room-wZQ7J42Y5SdMquaJ5 HTTP/1.1" 204 0 "-" "aws-sdk-nodejs/2.738.0 linux/v14.18.3 callback" 543 0.030 [minio-hot-minio-443] [] 10.244.30.17:9000 0 0.032 204 xx
xx - - [30/Mar/2022:13:21:28 +0000] "GET /chat-liiib-re/xx/avatars/room-wZQ7J42Y5SdMquaJ5?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=xx HTTP/2.0" 404 401 "-" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:96.0) Gecko/x Firefox/96.0" 301 0.013 [minio-hot-minio-443] [] 10.244.30.17:9000 401 0.012 404 xx
```

So it looks like there is a race condition on the room avatar, that sometimes it deletes it after uploading it, and sometimes not.

Hope it helps!
